apis/r/R/utils-coercions.R

```diff
@@ -0,0 +1,40 @@
+#' Coercion methods for SOMA classes
+
+#' @importFrom arrow as_arrow_table
+#' @export
+as_arrow_table.SOMASparseNDArrayRead <- function(x){
+  x$tables()$concat()
+}
+
+#' Coerce \link[tiledbsoma]{SOMASparseNDArrayRead} to \link{data.frame} or \link[tibble]{tibble}
+#' @export
+as.data.frame.SOMASparseNDArrayRead <- function(x, ...){ 
+   as.data.frame(x$tables()$concat(), ...)
+}
+
+# Coerce \link[tiledbsoma]{SOMASparseNDArrayRead} to Matrix::\link[Matrix]{dgTMatrix}
+setAs(from = "SOMASparseNDArrayRead", 
+      to = "TsparseMatrix", 
+      def = function(from) from$sparse_matrix()$concat()
+      )
```

Comment:
The R6 classes may need to be declared as an old class with `methods::setOldClass()`

```R
methods::setOldClass("SOMASparseNDArrayRead")
```
We may also need to import the target classes from Matrix
```R
#' @importClassesFrom Matrix TsparseMatrix CsparseMatrix RsparseMatrix 
#'
NULL
```